code/lib/builder-webpack5/src/index.ts

```diff
@@ -134,7 +133,7 @@ const starter: StarterFunction = async function* starterGeneratorFn({
   router.use(webpackHotMiddleware(compiler as any));
 
   const stats = await new Promise<Stats>((ready, stop) => {
-    compilation.waitUntilValid(ready as any);
+    compilation?.waitUntilValid(ready as any);
```

Comment:
This changes the behavior a little since the original would throw if `compilation` was undefined while this fails silently. But since `compilation` is being defined above I think that's okay.